### PR TITLE
Remove Style/IfUnlessModifier cop

### PR DESCRIPTION
**Pull request by kevin-j-m**
By default, rubocop will look for instances where an if test can be
rewritten to have a trailing conditional (or a modifier, as it calls
it). So, given the following code:

```
if true
  puts "hello!"
end
```

By default, rubocop will propose changing that to:

```
puts "hello!" if true
```

That raises readability concerns for multiple team members here, where
the condition at the end of the line may be passed over if glancing
quickly, and the explicit `if`, though more lines of code, can be a more
explicit and helpful form of the same expression.

As a result, this proposes turning that cop off so that it'll be up to
the best judgement of the developer for how to write their if
conditions. As far as I'm aware, there is no nuance or options available
with this [cop](https://rubocop.readthedocs.io/en/latest/cops_style/#styleifunlessmodifier), which leaves us with an on or off decision.

### CODE REVIEW

**Comment by kevin-j-m** [comment]
Personally, I like using trailing conditionals for guard clauses at the tops of methods, and I also have no strong aversion to them generally, but I know I've heard readability concerns from various folks here, so I want to make sure I'm capturing that correctly.

That being said, this change doesn't affect my ability to do that, it just removes rubocop from suggesting that in all cases where you can put it on one line.

**Review by danfrenette — APPROVED** [review]
As one such person voicing said readability concerns, I'd just like to explain that my aversion to using them mainly comes from cases where the code in question is at such a high length (as in physical length, measured in columns) that the if statement can't physically be seen, and thus the true complexity of the code becomes hidden. This is not uncommon in HTML files which are not checked by Rubocop for line length.

TL;DR I'm cool with trailing modifiers, just like, don't use them to sneak in weird stuff 300 columns outside of my viewer.

Approving because I'm super baised towards this change, but feel free to disagree!

**Review by mikestone14 — APPROVED** [review]
